RELEASES.md

```diff
@@ -1,3 +1,123 @@
+Version 1.61.0 (2022-05-19)
+==========================
+
+Language
+--------
+
+- [`const fn` signatures can now include generic trait bounds][93827]
+- [`const fn` signatures can now use `impl Trait` in argument and return position][93827]
+- [Function pointers can now be created, cast, and passed around in a `const fn`][93827]
+- [Recursive calls can now set the value of a function's opaque `impl Trait` return type][94081]
+
+Compiler
+--------
+
+- [Linking modifier syntax in `#[link]` attributes and on the command line, as well as the `whole-archive` modifier specifically, are now supported][93901]
+- [Update to LLVM 14.0.0][95247]
+- [The `char` type is now described as UTF-8 (`DW_ATE_UTF`) in DWARF debuginfo][89887]
```

Comment:
The more interesting part of this change is that `char` variables now properly display in Windows (msvc) debuggers. Prior to this change, variables of type `char` could not be inspected at all. It might be good to mention this.

Reply:
If it affects MSVC, then I guess this is not only about DWARF? Even though LLVM basically uses DWARF in IR, it will translate that as needed -- looks like CodeView gets a `char32_t` encoding [here](https://github.com/llvm/llvm-project/blob/9c8a88382d86c731db3c5c92d8ecd6ef296329ab/llvm/lib/CodeGen/AsmPrinter/CodeViewDebug.cpp#L1819-L1825).

Even for DWARF targets, `char` is not UTF-8, but UTF and a byte size attribute to represent UTF-32.

Reply:
Yeah, this is a bit confusing. LLVM's debuginfo annotations are basically just DWARF and map very closely to it. For Windows, which doesn't use DWARF at all and instead uses CodeView records embedded in PDBs, LLVM walks the debug info it has at the end of the compilation session and attempts to generate similar CV records from it. In many cases, there isn't a good way to map the DWARF style debuginfo into CV records and LLVM will silently discard the debuginfo in these cases.

Prior to this change, there was no mapping into a primitive CV type for `DW_ATE_unsigned_char` and so the debuginfo related to `char` variables was silently discarded resulting in them being completely unknown to the debugger. LLDB has a similar problem where it didn't know how to display `DW_ATE_unsigned_char` and would report an error to the user instead of showing the value.

There is however a mapping for `DW_ATE_UTF` to primitive CV types so this change now causes LLVM to be able to generate the appropriate debuginfo for `char`s and LLDB also knows to display the value.

Reply:
Maybe we should just leave out the specifics?
```suggestion
- [The `char` type is now described as UTF-32 in debuginfo][89887]
```